sdk/core/Azure.Core/src/RehydrationToken.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents a token that can be used to rehydrate a long-running operation.
+    /// </summary>
+    public readonly partial struct RehydrationToken
```

Comment:
How are we supporting mocking?   I think this needs to be added to a model factory - but, uh, Core doesn't have one.   @KrzysztofCwalina - should we be thinking about adding one?

Reply:
I think mocking can be handled through MRW.  You can hand craft any rehydration string and use MRW.Create to convert that into a RehydrationToken.

Reply:
That goes against the documented approach for mocking client types and would be really odd for that restriction to be applied to just this one type.

Reply:
I think its closer to the new Mocking strategy that @KrzysztofCwalina has been talking about.

I also don't think mocking this type will be something that any customers actually do.  If they do it would be an very advanced scenario.

Reply:
I would like to consider moving away from the model factory pattern in System.ClientModel based libraries in favor of something like that: https://github.com/KrzysztofCwalina/azure-sdk-for-net/blob/modelfactory/sdk/core/System.Net.ClientModel/tests/client/OpenAIClientTests.cs#L38

Reply:
If the above is not sufficient for users, we can always add factories later. But, I don't like the model factory pattern as it does not evolvevery well and after a couple of years, all the factory methods are a total mess.

Reply:
@KrzysztofCwalina / @m-nash:  Should we at least add a protected constructor to allow for mocking frameworks?   It could set a dummy Id so there's a value present.

Reply:
> I would like to consider moving away from the model factory pattern in System.ClientModel based libraries in favor of something like that: https://github.com/KrzysztofCwalina/azure-sdk-for-net/blob/modelfactory/sdk/core/System.Net.ClientModel/tests/client/OpenAIClientTests.cs#L38

Thanks for this example, @KrzysztofCwalina.   When @m-nash suggested MRW, I was picturing having to create JSON as the basis - the anonymous object approach is much clearer and more readable that what I thought was being suggested.   With the additional context, I no longer have concerns with the approach.

That said, if we are going to make this a formal pattern, we'll need to loop @scottaddie in so that we can add examples to the [Unit testing and mocking with the Azure SDK for .NET](https://learn.microsoft.com/dotnet/azure/sdk/unit-testing-mocking?tabs=csharp) article that illustrate.  We continue to see incoming isseus with a variation of "how can I mock these things" 2-3 times a month.

Reply:
Please keep me in the loop on what we decide. I'll engage the content team once a decision is reached.

Reply:
Yeah, I am not yet fully committed to this new pattern. But I would prefer to not ship anything in ClientModel that would clash with it if we endup making a bet on this new pattern. I would propose we ship as -is (as in this PR) and if people complain we can add mockability in the next release